### PR TITLE
Ported a missing SqlParameter constructor

### PR DESCRIPTION
**Pull request by joelbraun**
I've added a missing SqlParameter constructor per discussion on #17126. This is a direct port of the constructor located [here](https://msdn.microsoft.com/en-us/library/5a10hy4y(v=vs.110).aspx?cs-save-lang=1&cs-lang=csharp#code-snippet-2) from reference source located [here](https://github.com/dotnet/corefx/blob/master/src/shims/ApiCompatBaseline.netcoreapp.netfx461.txt). I tried to mirror my work off the similar PR #19544.

@danmosemsft had originally suggested that I attempt to port the rest of SqlParameter's missing items as well, but since this is my first attempt at open source contribution I'm going to start small. (Plus, some of that work looked a bit out of my league.)

I'm happy to make any changes or additions as necessary - please let me know!

### CODE REVIEW

**Comment by danmosemsft** [comment]
Q thanks @joelbraun ! Could you please add a test? They are in the tests folder next to the src folder. There are docs on how to run them, in the documentation folder.

**Comment by joelbraun** [comment]
Absolutely, I've added in an integration test and verified its functionality.

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SqlParameter.cs [hunk 1]
```diff
@@ -112,6 +112,31 @@ public SqlParameter(string parameterName, SqlDbType dbType, int size, string sou
...
+            SqlDbType dbType,
+            int size,
+            ParameterDirection direction,
+            bool isNullable,
+            byte precision,
+            byte scale,
+            string sourceColumn,
+            DataRowVersion sourceVersion,
+            object value
+        ) : this()
```
Comment (ViktorHofer) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SqlParameter.cs]: prefer using the constructor overload which already handles four parameters: `this(parameterName, dbType, size, sourceColumn)`

**Comment by danmosemsft** [comment]
@dotnet-bot test NETFX x86 Release Build please (disabled stack trace tests)

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SqlParameter.cs (hunk 1 above)
Comment (joelbraun) [reply src/System.Data.SqlClient/src/System/Data/SqlClient/SqlParameter.cs]: Agree with you on this, but I figured I'd go off the reference source. I've updated the PR to use that constructor instead.

**Comment by corivera** [comment]
@weshaggard Do we need to do a version bump after every contract change?

**Comment by corivera** [comment]
@dotnet-bot Test NETFX x86 Release Build please
@dotnet-bot Test Windows x86 Release Build please

**Comment by saurabh500** [comment]
Version bump is not needed for every contract change. If we have bumped once after 2.0 to reflect API additions then we are good.

**Comment by corivera** [comment]
@sausing The version was bumped when I added query notification support: https://github.com/dotnet/corefx/commit/27f42738d096a95419acbbf00f84c705536f795a

**Comment by joelbraun** [comment]
It looks like that last Windows x86 build is failing due to some kind of a timeout, or bigger problems.
`ERROR: The process "VBCSCompiler.exe" not found.`

**Comment by corivera** [comment]
@dotnet-bot Test Windows x86 Release Build please

**Comment by corivera** [comment]
Are we good to merge this?

File: src/System.Data.SqlClient/tests/ManualTests/SQL/ParameterTest/ParametersTest.cs [hunk 2]
```diff
@@ -98,6 +98,20 @@ public static void CodeCoverageSqlClient()
...
 
         [CheckConnStrSetupFact]
+        public static void Test_SqlParameter_Constructor()
+        {
+            using (var conn = new SqlConnection(s_connString))
+            {
+                conn.Open();
+                var cmd = new SqlCommand("select @input as 'test'", conn);
+                var sqlParameter = new SqlParameter("@input", SqlDbType.Int, 1, ParameterDirection.Input, false, 1, 0, "test", DataRowVersion.Current, MyEnum.A);
+                cmd.Parameters.Add(sqlParameter);
```
Comment (danmosemsft) [inline src/System.Data.SqlClient/tests/ManualTests/SQL/ParameterTest/ParametersTest.cs]: Can the test validate all the parameters -- eg., sourceColumn? 

Sorry to be picky!

**Comment by danmosemsft** [comment]
@corivera not sure who you're asking. as far as I know it is (notwithstanding test note above)

File: src/System.Data.SqlClient/tests/ManualTests/SQL/ParameterTest/ParametersTest.cs (hunk 2 above)
Comment (joelbraun) [reply src/System.Data.SqlClient/tests/ManualTests/SQL/ParameterTest/ParametersTest.cs]: Not a problem, I've updated the test to use an adapter and datatable to test sourceColumn and all other parameters. Not sure if there's a better way to do this, but it seemed like a thorough (and common) use case.

**Comment by danmosemsft** [comment]
@joelbraun entirely optional, but if you go to the <corefx>\src\shims folder and do `msbuild /p:baselineapicompat=true ApiCompat.proj` it will update the ApiCompatBaseline.netcoreapp.netfx461.txt file to remove this constructor you added!

It looks like it also removes some other lines, which is fine, it just is out of date.

Our goal is to clear out this file, as I mentioned in the issue...

**Comment by danmosemsft** [comment]
ARM boxes tracked by https://github.com/dotnet/dotnet-ci/pull/803. We can ignore them, I think.

**Comment by danmosemsft** [comment]
Actually, if you update the .txt file it will just reset CI again, we can do that later or this will never get in :)

**Comment by joelbraun** [comment]
Haha, sounds good. Glad to have helped a little bit with that file at least.

**Comment by corivera** [comment]
@dotnet-bot Test Linux arm Release Build please
@dotnet-bot Test Tizen armel Debug Build please

**Comment by danmosemsft** [comment]
I'm going to merge rather than wait fo them to fix the ARM boxes as I think the risk of affecting just ARM is very small.

**Comment by danmosemsft** [comment]
@joelbraun do you want to do more ? Probably it will go quicker next time

**Comment by joelbraun** [comment]
I'm definitely willing to give some other stuff on that list a shot - I think I've got a better feel for the process now.

**Pull request merged**